packages/admin-app/src/App.tsx

```diff
@@ -1,15 +1,37 @@
 import React from 'react';
 import { BrowserRouter as Router } from 'react-router-dom';
-import { DefaultLayout } from './layouts';
+import { ThemeProvider } from '@material-ui/core';
+import { AuthenticatedLayout } from './layouts';
+import { LoginPage } from './pages';
+import useToken from './useToken';
 
-import MainRoutes from './routes/MainRoutes';
+import AuthenticatedRoutes from './routes/AuthenticatedRoutes';
+import UnauthenticatedLayout from './layouts/UnauthenticatedLayout';
+import theme from './theme';
 
-const App: React.FunctionComponent = () => (
-    <Router>
-        <DefaultLayout>
-            <MainRoutes />
-        </DefaultLayout>
-    </Router>
-);
+const App: React.FunctionComponent = () => {
+    const { setToken, token } = useToken();
+
+    const getContent = () => {
+        const unauthenticated = !token;
+        if (unauthenticated) {
+            return (
+                <UnauthenticatedLayout>
+                    <LoginPage setToken={setToken} />
+                </UnauthenticatedLayout>
+            );
+        }
+
+        return (
+            <Router>
+                <AuthenticatedLayout>
+                    <AuthenticatedRoutes />
+                </AuthenticatedLayout>
+            </Router>
+        );
+    };
+
+    return <ThemeProvider theme={theme()}>{getContent()}</ThemeProvider>;
```

Comment:
It seemed odd that there was a theme, but it was never actually used, so I added it here at the root